File: GuiaApe06/Ejercicio7.java
import java.util.ArrayList; // Librería para usar ArrayList

public class Ejercicio7 {

    public static void main(String[] args) {

        // =========================
        // ARREGLO
        // =========================

        // Crear un arreglo fijo con 3 nombres
        String[] arreglo = {"Juan", "Pedro", "Maria"};

        // Mostrar título
        System.out.println("ARREGLO");

        // Recorrer y mostrar el arreglo
        for(String nombre : arreglo) {

            // Mostrar cada nombre
            System.out.println(nombre);
        }

        // =========================
        // ARRAYLIST
        // =========================

        // Crear un ArrayList dinámico
        ArrayList<String> lista = new ArrayList<>();

        // Agregar elementos al ArrayList
        lista.add("Juan");
        lista.add("Pedro");
        lista.add("Maria");
        lista.add("Carlos");

        // Mostrar título
        System.out.println("\nARRAYLIST");

        // Recorrer y mostrar el ArrayList
        for(String nombre : lista) {

            // Mostrar cada nombre
            System.out.println(nombre);
        }
    }
}

File: GuiaApe06/Ejercicio7.cpp
#include <iostream>   // Librería para entrada y salida de datos
#include <vector>     // Librería para usar vectores dinámicos

using namespace std;

int main() {

    // Crear un arreglo fijo con 3 nombres
    string arreglo[3] = {"Juan", "Pedro", "Maria"};

    // Mostrar título
    cout << "ARREGLO" << endl;

    // Recorrer y mostrar el arreglo
    for(int i = 0; i < 3; i++) {

        // Mostrar cada elemento del arreglo
        cout << arreglo[i] << endl;
    }

    // Crear un vector dinámico
    vector<string> lista;

    // Agregar elementos al vector
    lista.push_back("Juan");
    lista.push_back("Pedro");
    lista.push_back("Maria");
    lista.push_back("Carlos");

    // Mostrar título
    cout << "\nVECTOR" << endl;

    // Recorrer y mostrar el vector
    for(string nombre : lista) {

        // Mostrar cada nombre
        cout << nombre << endl;
    }

    return 0;
}
